streampipes-extensions/streampipes-sinks-notifications-jvm/src/main/java/org/apache/streampipes/sinks/notifications/jvm/msteams/MSTeamsSink.java

```diff
@@ -0,0 +1,227 @@
+                            Labels.withId(KEY_MESSAGE_ADVANCED),
+                            StaticProperties.stringFreeTextProperty(
+                                    Labels.withId(KEY_MESSAGE_ADVANCED_CONTENT),
+                                    true,
+                                    true
+                            )
+                    )
+            )
+            .build();
+  }
+
+  @Override
+  public void onInvocation(
+          SinkParams parameters,
+          EventSinkRuntimeContext runtimeContext
+  ) throws SpRuntimeException {
+    var extractor = parameters.extractor();
+    webhookUrl = extractor.secretValue(KEY_WEBHOOK_URL);
+
+    validateWebhookUrl(webhookUrl);
+
+    var selectedAlternative = extractor.selectedAlternativeInternalId(KEY_MESSAGE_TYPE_ALTERNATIVES);
+    if (selectedAlternative.equals(KEY_MESSAGE_ADVANCED)) {
+      isSimpleMessageMode = false;
+      messageContent = extractor.singleValueParameter(KEY_MESSAGE_ADVANCED_CONTENT, String.class);
+    } else {
+      isSimpleMessageMode = true;
+      messageContent = extractor.singleValueParameter(KEY_MESSAGE_SIMPLE_CONTENT, String.class);
+    }
+
+  }
+
+  @Override
+  public void onDetach() {
+    // nothing to do
+  }
+
+  /**
+   * Creates a JSON string intended for the MS Teams Webhook URL based on the provided plain message content.
+   * <p>
+   * This method utilizes a basic approach for constructing messages to be sent to MS Teams.
+   * If you intend to provide text in the form of Adaptive Cards, consider using
+   * {@link #createMessageFromAdvancedContent(String)} for a more advanced and interactive message format.
+   * </p>
+   *
+   * @param messageContent The plain message content to be included in the Teams message.
+   * @return A JSON string formatted using a predefined template with the provided message content.
+   */
+  protected String createMessageFromSimpleContent(String messageContent) {
+    return SIMPLE_MESSAGE_TEMPLATE.formatted(messageContent);
+  }
+
+  /**
+   * Creates a message for MS Teams from a JSON string, specifically designed for use with Adaptive Cards.
+   * <p>
+   * This method takes a JSON string as input, which is expected to represent the content of the message.
+   * The content is directly forwarded to MS Teams, allowing for the utilization of Adaptive Cards.
+   * Adaptive Cards provide a flexible and interactive way to present content in Microsoft Teams.
+   * Learn more about Adaptive Cards: <a href="https://learn.microsoft.com/en-us/adaptive-cards/">here</a>
+   * </p>
+   *
+   * @param messageContent The JSON string representing the content of the message.
+   * @return The original JSON string, unchanged.
+   * @throws JsonProcessingException If the provided message is not a valid JSON string.
+   */
+  protected String createMessageFromAdvancedContent(String messageContent) throws JsonProcessingException {
+    new ObjectMapper().readValue(messageContent, Object.class);
```

Comment:
Maybe we can use a class variable for the object mapper instead of instantiating it on every event

Reply:
Done